src/Processors/Transforms/WindowTransform.cpp

```diff
@@ -1157,7 +1187,7 @@ void WindowTransform::appendChunk(Chunk & chunk)
             // the frame boundaries, but it would require some care not to
             // perform unnecessary work while we are still looking for the frame
             // start, so do it the simple way for now.
-            updateAggregationState();
+            (this->*update_aggregate_state)();
```

Comment:
Maybe it will be the same performance with `if`, because it will be predictable, but it will be cleaner.